client/verta/verta/runtime.py

```diff
@@ -0,0 +1,227 @@
+# -*- coding: utf-8 -*-
+"""Classes and functions to enable context logging within predictions made against
+models on the Verta Platform.
+
+.. versionadded:: 0.22.0
+
+"""
+
+import json
+import re
+import threading
+from typing import Any, Dict, Optional
+
+
+_THREAD = threading.local()
+_S3_REGEX = re.compile("^[a-zA-Z0-9!_.*'()-]+(/[a-zA-Z0-9!_.*'()-]+)*$")
```

Comment:
[Explanation of regex pattern](https://regex101.com/r/EFTruv/1) which conforms to the "Safe characters" described by AWS in their [object key naming doc](https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-keys.html)

Reply:
Simplified already.